app/endpoints/raffle.py

```diff
@@ -360,11 +412,11 @@ async def buy_ticket(
         raise HTTPException(status_code=429, detail="Too fast !")
     locker_set(redis_client=redis_client, key=redis_key, lock=True)
 
-    new_amount = balance.balance - type_ticket.price
-    if new_amount < 0:
-        raise HTTPException(status_code=400, detail="Not enough cash")
-
     try:
+        new_amount = balance.balance - type_ticket.price
+        if new_amount < 0:
+            raise HTTPException(status_code=400, detail="Not enough cash")
```

Comment:
Raising a `HTTPException` in the try/catch block works as the error code is 400 in both cases but is a bit strange

Reply:
Replaced with a ValueError (which is also quite unusual in an endpoint?)